openslides_backend/action/actions/user/user_mixin.py

```diff
@@ -173,3 +186,42 @@ def get_history_information(self) -> Optional[HistoryInformation]:
                     fqid_from_collection_and_id("user", instance["id"])
                 ] = instance_information
         return information
+
+
+class DuplicateCheckMixin(Action):
+    def init_duplicate_set(self, data: List[Any]) -> None:
+        self.users_in_double_lists = self.execute_presenter(
+            SearchUsers,
+            {
+                "permission_type": "organization",
+                "permission_id": 1,
+                "search": data,
+            },
+        )
+        self.used_usernames: List[str] = []
+        self.used_names_and_email: List[Any] = []
+
+    def check_username_for_duplicate(self, username: str, payload_index: int) -> bool:
+        result = (
+            self.users_in_double_lists[payload_index] or username in self.used_usernames
+        )
+        if username not in self.used_usernames:
+            self.used_usernames.append(username)
+        return result
+
+    def check_name_and_email_for_duplicate(
+        self, first_name: str, last_name: str, email: str, payload_index: int
+    ) -> bool:
+        entry = (first_name, last_name, email)
+        result = (
+            self.users_in_double_lists[payload_index]
```

Comment:
what happens if you'll find 2 entries with different usernames (unique). Add a test for this case, even if it's difficult to appear in real life

Reply:
Two entries in the `users_in_double_lists[payload_index]` leads to not get the search_data and an Error in the state.